lib/signers/v4.js

```diff
@@ -11,10 +11,12 @@ var expiresHeader = 'presigned-expires';
  * @api private
  */
 AWS.Signers.V4 = inherit(AWS.Signers.RequestSigner, {
-  constructor: function V4(request, serviceName, signatureCache) {
+  constructor: function V4(request, serviceName, options) {
     AWS.Signers.RequestSigner.call(this, request);
     this.serviceName = serviceName;
-    this.signatureCache = signatureCache;
+    options = options || {};
```

Comment:
If someone were constructing a signer with `signatureCache = true`, options would still be a boolean here, so `options.signatureCache` would be undefined.